### PR TITLE
Add net46 and netcore50 configurations for System.Net.NetworkInformation

### PR DESCRIPTION
**Pull request by mellinoe**
All of the source code is already here, just adding the project configurations and appropriate project.json entries.

@davidsh , @CIPop , @weshaggard

### CODE REVIEW

File: src/System.Net.NetworkInformation/src/System.Net.NetworkInformation.csproj [hunk 1]
```diff
@@ -10,11 +10,13 @@
     <OutputType>Library</OutputType>
     <ProjectGuid>{3CA89D6C-F8D1-4813-9775-F8D249686E31}</ProjectGuid>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
-    <PackageTargetFramework>dotnet5.4</PackageTargetFramework>
+    <EnableWinRT Condition="'$(TargetGroup)' == 'netcore50'">true</EnableWinRT>
+    <PackageTargetFramework Condition="'$(TargetGroup)' == ''">dotnet5.4</PackageTargetFramework>
     <PackageTargetFramework Condition="'$(TargetsWindows)' == 'true' AND '$(EnableWinRT)' == 'true'">netcore50</PackageTargetFramework>
-    <PackageTargetRuntime Condition=" '$(TargetsWindows)' == 'true'">win7</PackageTargetRuntime>
+    <PackageTargetRuntime Condition=" '$(TargetsWindows)' == 'true' And '$(TargetGroup)' == ''">win7</PackageTargetRuntime>
```
Comment (mellinoe) [inline src/System.Net.NetworkInformation/src/System.Net.NetworkInformation.csproj]: @ericstj Could you take a look at these changes? I was getting package restore errors (from the net46 group) if I didn't change this.

**Comment by CIPop** [comment]
Thanks for helping us with the netcore50 port @mellinoe!
Do we need to worry about the tests at this point? (Tests will have to be different for netcore where we don't have many things implemented yet.)

LGTM otherwise

**Comment by davidsh** [comment]
>Do we need to worry about the tests at this point? (Tests will have to be different for netcore where we don't have many things implemented yet.)

We do have some tests for netcore50 but they are not ported to GitHub.  And currently there is no way to run on via GitHub CI anyways.  That infrastructure work is still ongoing for the next few months.

**Comment by davidsh** [comment]
LGTM

**Comment by stephentoub** [comment]
Test Innerloop Ubuntu Release Build and Test please
Test Innerloop CentOS7.1 Release Build and Test please
Test Innerloop CentOS7.1 Debug Build and Test please

**Comment by mellinoe** [comment]
> Do we need to worry about the tests at this point? 

To add on to what David said, testing will remain the same for now, We'll still be relying on the internal tests to continue running until we can move that infrastructure out here.

Tests passed, going to merge in now.

**Comment by weshaggard** [comment]
LGTM

**Pull request merged**